packages/doenetml-worker-rust/lib-doenetml-core/src/core/components/doenet/boolean.rs

```diff
@@ -12,35 +12,46 @@ pub struct Boolean {
     pub no_rendered_children: Vec<ComponentPointerTextOrMacro>,
 }
 
+impl Boolean {
+    /// If a Boolean component extends a state variable of type boolean,
+    /// then its value state variable should shadow that variable.
+    ///
+    /// Returns: a tuple of (component type, state variable index)
+    pub fn get_state_variable_that_shadows_when_extending() -> (&'static str, StateVarIdx) {
+        (
+            Boolean::get_component_type(),
+            BooleanStateVariables::get_value_state_variable_index(),
+        )
+    }
+}
+
 #[derive(Debug, ComponentStateVariables)]
 pub struct BooleanStateVariables {
+    #[is_public]
+    #[for_renderer]
+    #[component_profile_state_variables(Boolean)]
```

Comment:
docstring? Hopefully when we get enough of this working, I can look into extracting the docstrings to automatically create documentation :-)

Reply:
I cannot figure out how to add a docstring to derive macro help attributes! I could put the documentation on the `ComponentStateVariables` derive macro, but I'd really like to be able to hover over `#[for_renderer]` to see what that means.

Reply:
I'm confused. I meant to imply that `value` should have a doc string, not the macros themselves.

Reply:
OK. I added a docstring to `ComponentStateVariables` given that I misunderstood your request. Now added docstrings to the state variable fields.